src/analysis/retail/warlock/demonology/modules/features/SummonDemonicTyrant.tsx

```diff
@@ -1,143 +1,150 @@
+import Statistic from 'parser/ui/Statistic';
 import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
 
 import DemoPets from '../pets/DemoPets';
-import { isWildImp } from '../pets/helpers';
+import './SummonTyrant.scss';
+
+const debug = false;
 
 class SummonDemonicTyrant extends Analyzer {
   static dependencies = {
     demoPets: DemoPets,
   };
 
   demoPets!: DemoPets;
-  demonicTyrantPower!: number[];
+  _reignOfTyrannyPower!: number[];
   _hasReignOfTyranny!: boolean;
   _petsPerCast!: Record<number, number>[];
 
+  _summsWithDemonicPower!: { [id: string]: any }[];
+  _tyrantCast!: number;
+
   constructor(options: Options) {
     super(options);
-    this.demonicTyrantPower = [];
+    this._tyrantCast = 0;
+    this._summsWithDemonicPower = [];
+    this._summsWithDemonicPower[0] = {};
     this._hasReignOfTyranny = this.selectedCombatant.hasTalent(TALENTS.REIGN_OF_TYRANNY_TALENT);
-    this._petsPerCast = [];
+
     this.addEventListener(
       Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SUMMON_DEMONIC_TYRANT),
       this.summonDemonicTyrantCast,
     );
-    this._hasReignOfTyranny = this.selectedCombatant.hasTalent(TALENTS.REIGN_OF_TYRANNY_TALENT);
+    this.addEventListener(
+      Events.applybuff.to(SELECTED_PLAYER_PET).spell(SPELLS.DEMONIC_POWER),
+      this.demonicPowerAppliedToPet,
+    );
   }
 
   summonDemonicTyrantCast() {
-    const pets = this.demoPets.currentPets;
-    const countsPerCast = pets.reduce<Record<number, number>>((countsObj, currPet) => {
-      const { summonedBy } = currPet;
-      return { ...countsObj, [summonedBy]: (countsObj[summonedBy] || 0) + 1 };
-    }, {});
-    const demonicServitudeStacks = pets.reduce((total, currPet) => {
-      if (isWildImp(currPet.guid)) {
-        return total + 1;
-      } else {
-        return total + 2;
-      }
-    }, 0);
-    // pets.forEach((pet) => {
-    //   if (isWildImp(pet.guid)) {
-    //     demonicServitudeStacks += 1;
-    //   } else {
-    //     demonicServitudeStacks += 2;
-    //   }
-    //   countsPerCast[pet.summonedBy] = (countsPerCast[pet.summonedBy] || 0) + 1;
-    // });
-
-    this.demonicTyrantPower.push(demonicServitudeStacks);
-    this._petsPerCast.push(countsPerCast);
+    this._tyrantCast += 1;
+    this._summsWithDemonicPower[this._tyrantCast] = {};
   }
 
-  myList = [
-    { 1: 2, 3: 5, 6: 22 },
-    { 1: 1, 3: 5 },
-  ];
+  demonicPowerAppliedToPet(event: ApplyBuffEvent) {
+    if (!event.targetID) {
+      debug && this.error('Pet applyBuff event with no targetID', event);
+      return;
+    }
+    const petInfo = this.demoPets._getPetInfo(event.targetID);
+    if (!petInfo) {
+      debug && this.error(`Pet applyBuff event with nonexistant pet id ${event.sourceID}`);
+      return;
+    }
+
+    // TODO fix main pet getting two applications of this buff in the same tyrant
+    this._summsWithDemonicPower[this._tyrantCast][petInfo.name] =
+      this._summsWithDemonicPower[this._tyrantCast][petInfo.name] + 1 || 1;
+  }
```

Comment:
does this TODO mean that the main pet gets shown 2x on the statistic? or is the main pet mostly irrelevant for the stat?